UI/src/assetgrouping/asset-grouping.js

```diff
@@ -3,8 +3,14 @@
     var doGroupingByDistance = function(items, zoomLevel) {
       var result = [];
       var item;
-      var findProximityStops = function(x) {
-        return geometrycalculator.getSquaredDistanceBetweenPoints(x, item) < assetGroupingDistance;
+      var findProximityStops = function (x) {
+        if (x.id === item.id) {
+          return true;
+        } else if (x.floating || item.floating) {
+          return false;
+        } else {
+          return geometrycalculator.getSquaredDistanceBetweenPoints(x, item) < assetGroupingDistance;
+        }
       };
```

Comment:
Probably this check could be simplified

      if (x.floating || item.floating) {
          return false;
        } else {
          return geometrycalculator.getSquaredDistanceBetweenPoints(x, item) < assetGroupingDistance;
        }

x.id === item.id have the same point coordinates -> geometrycalculator.getSquaredDistanceBetweenPoints(x, item) < assetGroupingDistance; will resturn true

Reply:
I already try that approach and 'doesn't work.
This function (findProximityStops) it's using below with a "_.remove" ho will mutate the inserted array, because of that if we don't put the first part of the if, they never include himself on the array.